src/Native/Unix/configure.cmake

```diff
@@ -33,24 +33,33 @@ check_include_files(
     HAVE_LINUX_IN_H)
 
 if (HAVE_LINUX_IN_H)
-    set (SOCKET_INCLUDES ${SOCKET_INCLUDES} linux/in.h)
+    set (SOCKET_INCLUDES linux/in.h)
 else ()
-    set (SOCKET_INCLUDES ${SOCKET_INCLUDES} netinet/in.h)
+    set (SOCKET_INCLUDES netinet/in.h)
 endif ()
 
-set(CMAKE_EXTRA_INCLUDE_FILES ${SOCKET_INCLUDES})
-
-check_type_size(
-    "struct in_pktinfo"
-    HAVE_IN_PKTINFO
-    BUILTIN_TYPES_ONLY)
+check_c_source_compiles(
+    "
+    #include <${SOCKET_INCLUDES}>
+    int main()
+    {
+        struct in_pktinfo;
+        return 0;
+    }
+    "
+    HAVE_IN_PKTINFO)
```

Comment:
I ask any time we change tests in this file... have you verified these tests are correctly passing where they should and correctly failing where they shouldn't?